crmprtd/download_cache_process.py

```diff
@@ -0,0 +1,371 @@
+"""
+This script choreographs the download and process steps (with optional data caching)
+in the pipeline.
+
+TODO: More doc.
+"""
+
+import datetime
+from argparse import ArgumentParser
+from typing import List
+
+import pytz
+
+from crmprtd import NETWORKS, add_version_arg
+from crmprtd.argparse_helpers import OneAndDoneAction
+from crmprtd.infill import download_and_process
+
+# Maps a network alias to a *list* of networks (names). This allows for both groups
+# and single-network aliases.
+# TODO: This should probably be in a config file.
+network_aliases = {
+    "bch": ["bc_hydro"],
+    "hourly_swobml2": [
+        "bc_env_snow",
+        "bc_forestry",
+        "bc_tran",
+        "dfo_ccg_lighthouse",
+    ],
+    "ytnt": [
+        "nt_forestry",
+        "nt_water",
+        "yt_gov",
+        "yt_water",
+        "yt_avalanche",
+        "yt_firewx",
+    ],
+}
+
+network_alias_names = tuple(network_aliases.keys())
+
+
+def check_network_name(network_name):
+    if network_name not in NETWORKS:
+        raise ValueError(f"Network name '{network_name}' is not recognized.")
+
+
+def log_filename(
```

Comment:
On one hand, I'm hesitant to codify the output filepaths... but OTOH if a user requires flexibility, then they still have the ability to invoke the download scripts directly.

Reply:
We can allow the user to specify it with an optional log/cache filepath. If not specified, it can fall back to this.